### ISSUE TITLE
hlint incorrectly labels a "parse error"

### ISSUE THREAD

**Issue by ezzieyguywuf** [ndmitchell/hlint#1184]
With this minimal example:

```haskell
main :: IO (Maybe Bool)
main = do
    putStrLn "Hi"
    pure $do
        first <- Just True
        pure Nothing
```

I get the following output:

```sh
Error: Parse error
Found:
    main = do
        putStrLn "Hi"
>      pure $do
              first <- Just True
              pure Nothing
```

Adding a space before the second `do`, such that the marked line reads `pure $ do` fixes the issue.

(I realize the `pure Nothing` may be a type error, I'm still pretty new to haskell so let me know and I can fix the example).

I'm happy to provide any information that may help to determine if this is a real bug or not, and that may help in resolving it.

**Comment by shayne-fletcher** [comment]
HLint enables the extension `TemplateHaskell` by default and that's affecting the parse. You can get HLint to process the file with no hints by passing it the flag `-XNoTemplateHaskell`.

**Comment by ezzieyguywuf** [comment]
That does indeed fix it, thank you.

However, in the process I discovered that hlint will accept a bad command-line flag silently, i.e. `hlint -XNoTempateHaskell` ran just fine, with no errors, but there is a typo in the flag. Is this a bug?

**Comment by ndmitchell** [comment]
I believe this got fixed in #1094 as part of release 3.2. What does `--version` say?

**Comment by ezzieyguywuf** [comment]
Yikes, I'm on 2.2.11, I'll update and re-examine my life decisions. Thank you!
